frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx

```diff
@@ -170,24 +187,45 @@ export const EmbedFrame = ({
       data-testid="embed-frame"
       data-embed-theme={theme}
     >
-      <ContentContainer>
+      <ContentContainer id={DASHBOARD_PDF_EXPORT_ROOT_ID}>
         {hasHeader && (
           <Header
-            className={EmbedFrameS.EmbedFrameHeader}
+            className={cx(
+              EmbedFrameS.EmbedFrameHeader,
+              SAVING_DOM_IMAGE_DISPLAY_NONE_CLASS,
+            )}
             data-testid="embed-frame-header"
           >
-            {finalName && (
+            {(finalName || downloads) && (
               <TitleAndDescriptionContainer>
-                <FixedWidthContainer
+                <TitleAndButtonsContainer
                   data-testid="fixed-width-dashboard-header"
                   isFixedWidth={dashboard?.width === "fixed"}
                 >
-                  <TitleAndDescription
-                    title={finalName}
-                    description={description}
-                    className={CS.my2}
-                  />
-                </FixedWidthContainer>
+                  {finalName && (
+                    <TitleAndDescription
+                      title={finalName}
+                      description={description}
+                      className={CS.my2}
+                    />
+                  )}
+                  <Box style={{ flex: 1 }} />
+                  {/* TODO: move this to a prop passed by PublicDashboard ? */}
+                  {dashboard && downloads && (
+                    <Button
+                      variant="subtle"
+                      leftIcon={<Icon name="document" />}
+                      color="text-dark"
+                      onClick={saveAsPDF}
+                    >
+                      {/* TODO: refactor into a util as it's duplicated logic */}
```

Comment:
This logic for the text is also used for the download button for normal dashboards, I'll clean this up